Expensify/App
# Fix: Size of the single emoji in chat along with updated Emoji Regex

@deetergp I've updated my changes with the regex changes and a unit test to verify all the emojis in our dataset.

### Details
- Changed `isSingleEmoji` function
- Replaced Emoji Regex

### Fixed Issues

$ https://github.com/Expensify/App/issues/4543 

### Tests
1. Tested different variants of the Emojis with single length and surrogate pairs
2. Tested emojis with skin tone.
3. Tested with composite emojis such as Family
4. Tested text-based Emojis 



### QA Steps
1. Go to any Chat
2. Enter one emoji per message.
3. Size of the Emoji should be larger similar to other apps like Whatsapp, etc.
4. It should work for all variations.



### Tested On


### Screenshots


#### Web
<img width="642" alt="Screenshot 2021-09-08 at 5 47 53 PM" src="https://user-images.githubusercontent.com/3069065/132509116-6eca3a46-7173-4867-9e2e-d609f66bc60b.png">


#### Mobile Web

<img width="554" alt="Screenshot 2021-09-08 at 5 47 43 PM" src="https://user-images.githubusercontent.com/3069065/132509120-53b14583-9595-458c-81b8-222b558c96c3.png">
<img width="533" alt="Screenshot 2021-09-08 at 5 47 29 PM" src="https://user-images.githubusercontent.com/3069065/132509127-c3e17c75-b4a3-4798-bfca-3416fa573b03.png">
<img width="577" alt="Screenshot 2021-09-08 at 5 47 19 PM" src="https://user-images.githubusercontent.com/3069065/132509130-6dc1089f-25fd-4889-99e1-075590e271ce.png">

#### Desktop
<img width="935" alt="Screenshot 2021-09-08 at 6 00 29 PM" src="https://user-images.githubusercontent.com/3069065/132509467-a998f4fc-61f1-43f8-8105-2a11ccf50e5c.png">
<img width="864" alt="Screenshot 2021-09-08 at 6 00 20 PM" src="https://user-images.githubusercontent.com/3069065/132509479-e01a91b5-62a2-400d-9bd3-1250ac058538.png">


#### iOS
<img width="286" alt="Screenshot 2021-09-08 at 5 55 51 PM" src="https://user-images.githubusercontent.com/3069065/132509330-7c778a36-3e0b-4974-b490-63a899904d4c.png">
<img width="323" alt="Screenshot 2021-09-08 at 5 55 37 PM" src="https://user-images.githubusercontent.com/3069065/132509337-2585f255-9c3a-49b7-9efb-66841b4d04d8.png">


#### Android
<img width="363" alt="Screenshot 2021-09-08 at 6 08 21 PM" src="https://user-images.githubusercontent.com/3069065/132511301-819f2d6e-01f5-48c5-964c-1039c2aa6ba0.png">
<img width="354" alt="Screenshot 2021-09-08 at 6 08 09 PM" src="https://user-images.githubusercontent.com/3069065/132511312-90264e5d-0443-4070-b455-3db3f213a675.png">
<img width="374" alt="Screenshot 2021-09-08 at 6 07 54 PM" src="https://user-images.githubusercontent.com/3069065/132511321-8b25c99b-5489-45df-b4c7-f97cbe153b6c.png">
<img width="345" alt="Screenshot 2021-09-08 at 6 07 36 PM" src="https://user-images.githubusercontent.com/3069065/132511323-4807e11c-c6e3-4d73-9ec7-cf938f54da72.png">
